--- torsmo-0.16-orig/torsmo.c	2004-05-21 00:14:44.675856248 +0200
+++ torsmo-0.16/torsmo.c	2004-05-21 00:17:02.518900920 +0200
@@ -910,13 +910,13 @@
       get_load_average(v);
 
       if (obj->data.loadavg[2])
-        snprintf(p, n, "%.2f %.2f %.2f", v[obj->data.loadavg[0]-1],
-            v[obj->data.loadavg[1]-1], v[obj->data.loadavg[2]-1]);
+        snprintf(p, n, "%.2f %.2f %.2f", v[obj->data.loadavg[0]],
+            v[obj->data.loadavg[1]], v[obj->data.loadavg[2]]);
       else if (obj->data.loadavg[1])
-        snprintf(p, n, "%.2f %.2f", v[obj->data.loadavg[0]-1],
-            v[obj->data.loadavg[1]-1]);
+        snprintf(p, n, "%.2f %.2f", v[obj->data.loadavg[0]],
+            v[obj->data.loadavg[1]]);
       else if (obj->data.loadavg[0])
-        snprintf(p, n, "%.2f", v[obj->data.loadavg[0]-1]);
+        snprintf(p, n, "%.2f", v[obj->data.loadavg[0]]);
     }
     OBJ(hr) {
       new_hr(p, obj->data.i);
